### PR TITLE
removed connect fn, and now using useSelector and useDispatch

### PR DESCRIPTION
**Pull request by kimheffy**
Line 21 on TabularView.js, I feel like I can call `useSelector` once, and assign the return values to `changeToRatingHeadline and changeToQuoteHeadline`. Instead of calling `useSelector` twice for the respective variables (changeToRatingHeadline and changeToQuoteHeadline).

### CODE REVIEW

File: src/tabular-view/TabularView.js [hunk 1]
```diff
@@ -13,24 +13,18 @@ import Quote from '../quote/Quote';
...
-  changeToRatingHeadline: () => dispatch(changeTabularHeadline('rating')),
-  changeToQuoteHeadline: () => dispatch(changeTabularHeadline('quote'))
-});
+  // dispatch actions
+  const changeToRatingHeadline = () => dispatch(changeTabularHeadline('rating'));
+  const changeToQuoteHeadline = () => dispatch(changeTabularHeadline('quote'));
+
+  // useSelectors
+  const currentHeadline = useSelector((state) => state.tabularReducer.currentHeadline);
+  const quoteResult = useSelector((state) => state.tabularReducer.quoteResult);
```
Comment (kimheffy) [inline src/tabular-view/TabularView.js:26]: Right here is what I am talking about in the conversation.

File: src/app/App.js [hunk 2]
```diff
@@ -9,15 +9,13 @@ import './App.css';
...
-});
-
 // I would think the spinner should at this level
-// Because I can make the TabularView transparent
+// Because I can make the TabularView transparent(opacity)
 
 // eslint-disable-next-line react/prop-types
-function App({ isFetching }) {
+function App() {
+  const isFetching = useSelector((state) => state.tabularReducer.isFetching);
```
Comment (changyeamoon) [inline src/app/App.js:17]: can we separate the callback function to its own file?

File: src/tabular-view/TabularView.js (hunk 1 above)
Comment (changyeamoon) [reply src/tabular-view/TabularView.js:26]: i dont see why not. look up `reselect` and try to use that too

File: src/app/App.js (hunk 2 above)
Comment (kimheffy) [reply src/app/App.js:17]: i will give it a shot. and i'm guessing you want the callback fn into a separate file for? modularity?

File: src/tabular-view/TabularView.js (hunk 1 above)
Comment (kimheffy) [reply src/tabular-view/TabularView.js]: resolved it in latest commit

File: src/app/App.js [hunk 3]
```diff
@@ -9,25 +9,22 @@ import './App.css';
...
+import { mapIsFetchingFromState } from './AppMapSelector';
 
 // I would think the spinner should at this level
-// Because I can make the TabularView transparent
+// Because I can make the TabularView transparent(opacity)
 
 // eslint-disable-next-line react/prop-types
-function App({ isFetching }) {
+function App() {
+  const isFetching = useSelector(mapIsFetchingFromState);
```
Comment (kimheffy) [inline src/app/App.js:20]: callback function now resides in its own file

File: src/tabular-view/TabularView.js [hunk 4]
```diff
@@ -13,24 +13,26 @@ import Quote from '../quote/Quote';
...
+function TabularView({ toggleOpacity }) {
+  // dispatch
+  const dispatch = useDispatch();
+
+  // dispatch actions
+  const changeToRatingHeadline = () => dispatch(changeTabularHeadline('rating'));
+  const changeToQuoteHeadline = () => dispatch(changeTabularHeadline('quote'));
+
+  // useSelectors
+  const { currentHeadline, quoteResult } = useSelector((state) => ({
```
Comment (kimheffy) [inline src/tabular-view/TabularView.js:31]: i could've put this callback function into `./TabularMapSelectors`, but wrote it out here for explicitly and readability.

File: src/app/AppMapSelector.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export const mapIsFetchingFromState = (state) => state.tabularReducer.isFetching;
```
Comment (changyeamoon) [inline src/app/AppMapSelector.js:1]: usually selectors like these are grouped with the given store/reducer they are selecting from. 

Imagine if Quote.js wanted to use this selector. does it make sense to grab it from the App directory?

File: src/quote/QuoteMapSelector.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export const mapQuoteFromState = (state) => state.tabularReducer.quoteResult;
```
Comment (changyeamoon) [inline src/quote/QuoteMapSelector.js:1]: common naming convention is either `selectQuoteResult` or `QuoteResultSelector`

`mapQuoteFromState` sounds like its going through and changing quote from state

Comment (changyeamoon) [reply src/quote/QuoteMapSelector.js:1]: also same comment from above

File: src/tabular-view/TabularView.js (hunk 4 above)
Comment (changyeamoon) [reply src/tabular-view/TabularView.js:31]: thats fine for now

File: src/quote/QuoteMapSelector.js (hunk 6 above)
Comment (kimheffy) [reply src/quote/QuoteMapSelector.js:1]: thanks, I will get that fixed tomorrow morning

File: src/app/AppMapSelector.js (hunk 5 above)
Comment (kimheffy) [reply src/app/AppMapSelector.js:1]: i have failed to catch the pattern of reusability(selectors) from the start.

**Pull request merged**